[BLOG] Add Making pygments accessible (#527)
Quansight/Quansight-website: apps/labs/posts/lets-make-pygments-accessible.md

---
title: 'Let's make Pygments accessible!'
author: stephannie-jimenez

> published: November 7, 2022

Reviewer: ```suggestion
published: November 15, 2022
```

Moving a week later as we are deploying on Tuesdays and have a new post for this week

description: 'accessible-pygments hosts curated WCAG-compliant themes for all your syntax highlighting needs.'
category: [Access-centered]
featuredImage:

> src: /posts/lets-make-pygments-accessible/featured_themes.png

Reviewer: ```suggestion
  src: /posts/lets-make-pygments-accessible/featured_themes.png
```

alt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
hero:
imageSrc: /posts/lets-make-pygments-accessible/hero_themes.png

> imageAlt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'

Reviewer: ```suggestion
  imageAlt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'
```

---

> title: '"Let's make Pygments accessible!"'

Reviewer 2: ```suggestion
title: "Let's make Pygments accessible!"
```

Reviewer 2: I think the double quotes might be the issue here.

Reviewer: It was with single quotes before so will need inspecting closer

Reviewer 2: Sorry, poorly phrased! I meant double as in double within single.

Reviewer 2: The author slug is correct. I checked that before commenting. This is what made me investigate the row above to begin with.

Reviewer: Yup we have tried few combos of the quotes. 

@isabela-pf and @steff456  since it has been too much faff debugging this what about changing the title to "Making pygments accessible" 🙏🏽 ? At least to avoid delaying this once more

author: stephannie-jimenez-gacha
published: October 31, 2022
description: 'accessible-pygments hosts curated WCAG-compliant themes for all your syntax highlighting needs.'
category: [Access-centered]
featuredImage:

> src: /posts/lets-make-pygments-accessible/featured_themes.gif

Reviewer: Per @isabela-pf comment elsewhere, this needs updating for the static version

>   alt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'

Reviewer: ```suggestion
  alt: 'A display of different accessible syntax highlighting themes being applied to the same example code.'
```

since it is not an animation anymore

hero:
  imageSrc: /posts/lets-make-pygments-accessible/hero_themes.png

>   imageAlt: 'An animation of different accessible syntax highlighting themes being applied to the same example code.'

Reviewer: ```suggestion
  alt: 'A display of different accessible syntax highlighting themes being applied to the same example code.'
```

since it is not an animation anymore

---

Additional authors: Isabela Presedo-Floyd

Themes… everyone has their personal preferences when it comes to colors and fonts. Because let’s face it, we all have spent some time trying to change or create our own theme for our operating system, terminals, and IDEs. But what happens when preferences create inaccessible tools? And how do you fix it?

Creating themes is HARD enough, and checking whether a theme has proper contrast may be unfamiliar to people less aware of accessibility needs and standards. Additionally, there is no “correct” answer, so we need multiple theming options.

> [Pygments](https://pygments.org/) is a very popular package for syntax highlighting and is the default backend for a lot of packages. Within the Jupyter ecosystem, a community [we have worked to improve a breadth of accessibility issues](https://jupyter-a11y.netlify.app/) in, also uses Pygments as its syntax highlighting backend for projects like [Jupyter Book](https://jupyterbook.org/en/stable/intro.html). This means that in interacting with code cells and blocks, you are interacting with potential color issues inherited throughout the ecosystem. Knowing that Jupyter projects are far from the only ones that could benefit from accessibility-considerate syntax highlighting, we created our own package named `accessible-pygments` to host themes that already had accessible and WCAG compliant colors. To date, this includes 16 brand new themes based on [Eric Bailey’s a11y-syntax-highlighting themes](https://github.com/ericwbailey/a11y-syntax-highlighting), [github vscode themes](https://github.com/primer/github-vscode-theme), [gotthard vscode theme](https://github.com/janbiasi/vscode-gotthard-theme/), [blinds theme](https://github.com/orbulant/blinds-theme), [greative](https://github.com/SumanKhdka/Greative-VSCode-Theme) and [Tania Allard’s pitaya smoothie theme](https://github.com/trallard/pitaya_smoothie).

Reviewer: ```suggestion
[Pygments](https://pygments.org/) is a very popular package for syntax highlighting and is the default backend for many packages. Within the Jupyter ecosystem, a community [we have worked to improve a breadth of accessibility issues](https://jupyter-a11y.netlify.app/) in, also uses Pygments as its syntax highlighting backend for projects like [Jupyter Book](https://jupyterbook.org/en/stable/intro.html). This means that in interacting with code cells and blocks, you are interacting with potential color issues inherited throughout the ecosystem. Knowing that Jupyter projects are far from the only ones that could benefit from accessibility-considerate syntax highlighting, we created our package named `accessible-pygments` to host themes that already had accessible and WCAG-compliant colors. To date, this includes 16 brand new themes based on [Eric Bailey’s a11y-syntax-highlighting themes](https://github.com/ericwbailey/a11y-syntax-highlighting), [GitHub VSCode themes](https://github.com/primer/github-vscode-theme), [Gotthard VSCode theme](https://github.com/janbiasi/vscode-gotthard-theme/), [blinds theme](https://github.com/orbulant/blinds-theme), [greative](https://github.com/SumanKhdka/Greative-VSCode-Theme) and [Tania Allard’s Pitaya Smoothie theme](https://github.com/trallard/pitaya_smoothie).
```

The best accessibility experiences are often ones with strong defaults that welcome customization. In `accessible-pygments`, we are still focused on the strong defaults portion of that statement. Because we do not support customization, our goal is to [provide a growing list of curated accessible themes](https://github.com/Quansight-Labs/accessible-pygments/issues/2) for people to choose from. Having options so that everyone can choose what works best for them allows people to have agency and limit how disabling their environment is.

> As a note: general accessibility advice is to not rely exclusively on color to convey any information. Since syntax highlighting is all color information the authors have yet to find if there is an alternate approach.

Reviewer 3: But syntax highlighting doesn't convey any information that isn't already in the text itself. All it does it make the code easier to read by distinguishing the different types of code tokens.

Author: This is true! No information is lost by not being able to make use of syntax highlighting for any reason, but the kind of support that it can provide for someone reading code is lost as far as I'm aware. I probably didn't make that distinction clear enough.

Do you want me to remove the note in line 46?

Reviewer: Agree - no information is lost - _what does get affected though_ is more aligned with readability/scannability (if that is a word). 
Though - some IDE's and syntax highlighters _do_ allow the use of _italics_ to complement the "conveying information with colour"
